rllib/env/tests/conftest.py

```diff
@@ -0,0 +1,563 @@
+
+
+@pytest.fixture
+def env_runner_with_module_to_env_tracker(runner_type, ray_init):
+    """Create an EnvRunner with ModuleToEnvConnectorTracker installed."""
+    ModuleToEnvConnectorTracker.reset()
+
+    if runner_type == "single_agent":
+        from ray.rllib.algorithms import PPOConfig
+
+        config = (
+            PPOConfig()
+            .environment("CartPole-v1")
+            .env_runners(
+                num_envs_per_env_runner=1,
+                module_to_env_connector=make_module_to_env_connector_tracker,
+            )
+        )
+        runner = SingleAgentEnvRunner(config=config)
+    elif runner_type == "multi_agent":
+        from ray.rllib.algorithms import PPOConfig
+
+        config = (
+            PPOConfig()
+            .environment(MultiAgentCartPole, env_config={"num_agents": 2})
+            .multi_agent(
+                policies={"p0", "p1"},
+                policy_mapping_fn=lambda aid, *a, **kw: f"p{aid}",
+            )
+            .env_runners(
+                num_envs_per_env_runner=1,
+                module_to_env_connector=make_module_to_env_connector_tracker,
+            )
+        )
+        runner = MultiAgentEnvRunner(config=config)
+    else:
+        raise ValueError(f"Unknown runner type: {runner_type}")
+
+    yield runner
+    runner.stop()
+    ModuleToEnvConnectorTracker.reset()
+
+
+class ImmediateTerminationEnv(gym.Env):
+    """Environment that terminates immediately after one step."""
+
+    def __init__(self, config=None):
+        self.observation_space = gym.spaces.Discrete(2)
+        self.action_space = gym.spaces.Discrete(2)
+
+    def reset(self, *, seed=None, options=None):
+        return 0, {}
+
+    def step(self, action):
+        return 0, 1.0, True, False, {}
+
+
+class ZeroRewardEnv(gym.Env):
+    """Environment that always returns zero reward."""
+
+    def __init__(self, config=None):
+        self.observation_space = gym.spaces.Discrete(2)
+        self.action_space = gym.spaces.Discrete(2)
+        self._step_count = 0
+        self._max_steps = config.get("max_steps", 10) if config else 10
+
+    def reset(self, *, seed=None, options=None):
+        self._step_count = 0
+        return 0, {}
+
+    def step(self, action):
+        self._step_count += 1
+        done = self._step_count >= self._max_steps
+        return 0, 0.0, done, False, {}
+
+
+class NegativeRewardEnv(gym.Env):
+    """Environment that always returns negative reward."""
+
+    def __init__(self, config=None):
+        self.observation_space = gym.spaces.Discrete(2)
+        self.action_space = gym.spaces.Discrete(2)
+        self._step_count = 0
+        self._max_steps = config.get("max_steps", 10) if config else 10
+
+    def reset(self, *, seed=None, options=None):
+        self._step_count = 0
+        return 0, {}
+
+    def step(self, action):
+        self._step_count += 1
+        done = self._step_count >= self._max_steps
+        return 0, -1.0, done, False, {}
```

Comment:
I think we can probably get rid of `NegativeRewardEnv `, `ZeroRewardEnv` and `ImmediateTerminationEnv` and `ConfigurableFailingEnv`, right? I don't see them being used.